tests/release/releaseLib.go

```diff
@@ -77,6 +81,115 @@ func CreateComponent(devFw framework.Framework, devNamespace, appName, compName,
 	return component
 }
 
+// CreateComponentWithNewBranch will create a new branch, then create the component based on the new branch
```

Comment:
Why? Why can't we just use the head of the default branch? Why do we need a new branch to trigger an on-push build?

Reply:
Creating a new branch will make the test repeatable. It creates a PaC PR and merges it in each test and the branches will be cleaned in the end of the test. If we use the head of default branch, confilcts may happen in parallel runs.

Reply:
The RELEASE CR is created only by  an "on-push" event which is generated by merging a PR.  Adding comment of `/retest` can't generate an "on-push" event.

Reply:
Are you sure about that? That doesn't make sense to me. I am not talking about `/retest` on a PR. Like this https://github.com/konflux-ci/release-service/commit/6edf44528b6f2918b7625080d2cf888c6a040376#commitcomment-151860536

Reply:
In PaC config file, it usually monitors "event=push" to a specific branch which is configable by the user.  I checked that "git --force push" to an existing PR with the branch as target, it created a "pull-request" event.  I don't understand the link you pasted. Do you mean the comment of "/test" triggered a push event?

Reply:
Yes, that is what I mean. It started a new build pipelineRun for me

Reply:
A "pull request" event also trigger a build pipelineRun, but it doesn't create Release CR.

Reply:
I'm not sure what that comment is supposed to be getting at. My whole thing was I don't see why we need a pull request event

Reply:
We need a "push" event to create a Release CR and trigger the release pipeline at last. We need a new branch each time since we can't create a component for one test run with the same branch of a repo twice at the same time. The function is to create a new branch, then create a component each time.

Reply:
> we can't create a component for one test run with the same branch of a repo twice at the same time
Why?

Reply:
If you use the same branch to create a second component, it will tell you something like "it's already onboarded" in the status of Component CR. Konflux works in this way currently .

Reply:
Okay, so this is what my current understanding is:

- Your PR adds a pull request event *and* push
- We can't use the same branch twice

So why not just use a new branch for each test? What is the reason for the pull request? Wouldn't it work to just create a branch for the e2e run then delete it once it finishes?

Reply:
The pull request event (PaC init PR)  is created automatically. But we need a push event. So, the build pipelineRun is deleted and merge the pull request to create a push event.

Reply:
Creating a new branch is already a push, no? What is creating a pull request automatically?